cmd/cli/app/repo/repo_register.go

```diff
@@ -125,8 +126,15 @@ func getUnregisteredInputRepos(inputRepoList string, alreadyRegisteredRepos sets
 
 func fetchRemoteRepositoriesFromProvider(ctx context.Context, provider, project string, client minderv1.RepositoryServiceClient) (
 	[]*minderv1.UpstreamRepositoryRef, error) {
+	var provPtr *string
+	if provider != "" {
+		provPtr = &provider
+	}
```

Comment:
Can we make `""` also mean "choose any provider"?

Reply:
(If we can do that, we might not need to change the client at all)

Reply:
That is already the case with this PR. I'm anyway refining the approach.

Reply:
I was suggesting that we don't need to make the changes on 129-137, and we can just pass a pointer to `""` to the server, which would be less code and the same meaning.